File: MTADroneService_client/src/app/models/drone-model.ts
import { DroneCoordinatesModel } from "./drone-coord-model";

export class DroneModel{
    droneID:string="";
    droneModel:string="";
    droneStatus:string="";
    currentDroneCoordinates:DroneCoordinatesModel;

    constructor(private id:string, private model:string, private status:string, private coordinates:DroneCoordinatesModel){
        this.droneID = id;
        this.droneModel = model;
        this.status = status;
        this.currentDroneCoordinates = coordinates;
    }

    getDroneCoordinates(){
        return this.getDroneCoordinates;
    }
    
}

File: MTADroneService_server/DroneService/src/main/java/MTADroneService/DroneService/application/models/DroneModel.java
package MTADroneService.DroneService.application.models;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.mongodb.core.index.IndexDirection;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

/**
 * Class defining the drone model.
 *
 * @author Chirita Gabriela
 */
@Data
@Document(collection = "drone")
@NoArgsConstructor
@AllArgsConstructor
public class DroneModel {
    /**
     * Member description
     */
    @Indexed(unique = true)
    String droneID;
    String droneModel;
    String droneStatus;
}
